### PR TITLE
Free up some disk space before running `clang-tidy`

### PR DESCRIPTION
**Pull request by lnicola**
## Description

Might help with failures like https://github.com/qgis/QGIS/actions/runs/6493229254/job/17638277259.

### CODE REVIEW

**Review by strk — APPROVED** [review]
Took ~1 minute to saves 32gb:
https://github.com/qgis/QGIS/actions/runs/6506911127/job/17676861197?pr=54933#step:3:338

Build times overall seem so different that's hard to tell if that additional minute could ever be noticeable (we go from 2h to 4h and the buidl for this PR is on the "quick" side):
https://github.com/qgis/QGIS/actions/workflows/run-tests.yml?query=QGIS+tests

I guess it's good to reduce chances of failure so approving and merging.

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -442,6 +442,14 @@ jobs:
             qt-version: 5
 
     steps:
+      - name: Free Disk Space (Ubuntu)
+        uses: jlumbroso/free-disk-space@main
```
Comment (lnicola) [inline .github/workflows/run-tests.yml:446]: This is selected using the tag in other places too, but don't blame me if it ever gets compromised and the QGIS secrets get stolen.

Comment (strk) [reply .github/workflows/run-tests.yml:446]: Personally I hope no QGIS secrets are ever put in the github.com (microsoft) hands

Comment (lnicola) [reply .github/workflows/run-tests.yml:446]: These are: https://github.com/qgis/QGIS/blob/master/.github/workflows/build-docker.yml#L46-L48.

**Comment by strk** [comment]
In any case ***@***.*** was already used
before merging this PR. At least since August
( aca9bce6178c5ad7b08c8695f957a8cfa8046a9d )

**Pull request merged**